Fix captcha title being applied to menu items in non-block based themes [MAILPOET-6485]

NeosinneR opened the pull request:
## Description

On non-block-based theme, the Captcha title was applied to all titles on the page (those in the menu, widgets, ...) instead of just the main one (the one "in the loop").

| Before | After |
| ------ | ----- |
| <img width="735" alt="Screenshot 2025-02-19 at 09 12 44" src="https://github.com/user-attachments/assets/a90a5e84-43bb-4184-806b-d6675d8f09b3" /> | <img width="760" alt="Screenshot 2025-02-19 at 09 12 27" src="https://github.com/user-attachments/assets/93b70eb0-a757-48d2-9064-95c040deafd5" /> | 

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-6485]

## After-merge notes

_N/A_

## Tasks

- [x] 🚫 I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] 🚫 I added sufficient test coverage
- [x] 🚫 I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-6485]: https://mailpoet.atlassian.net/browse/MAILPOET-6485?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:captcha-page-title)

_The latest successful build from `captcha-page-title` will be used. If none is available, the link won't work._

oz reviewed (dismissed):
LGTM

Aschepikov commented:
Hi, @NeosinneR  With some themes there was no captcha title, can you check if it is ok for them?
![image](https://github.com/user-attachments/assets/957cd22f-c2d0-4219-90f2-8e42cd112739)
/wp-admin/themes.php?theme=silverstorm
![image](https://github.com/user-attachments/assets/fc33b1ff-a1ad-4847-add2-cc20a9894b09) 
/wp-admin/themes.php?theme=rufous

NeosinneR committed:
Fix captcha title being applied to menu items in non-block based themes

[MAILPOET-6485]

NeosinneR committed:
Fix showing Captcha title in themes that rely on single_post_title, such as Rufous or Silverstorm

[MAILPOET-6485]

NeosinneR commented:
@Aschepikov nice catch! Fixed in https://github.com/mailpoet/mailpoet/pull/6110/commits/51fc113573462fd3fbea0464a9fff1fd3c0e6b5c. I also changed the original fix, as it still replaced titles e.g. if the page included list of other pages in the content.

Aschepikov reviewed (approved):
GJ!
